pkg/reconciler/trigger/trigger.go

```diff
@@ -207,6 +240,37 @@ func (r *Reconciler) reconcile(ctx context.Context, t *v1alpha1.Trigger) error {
 	return nil
 }
 
+func (r *Reconciler) triggerMarkedReadyByDependency(dependencyObjRef corev1.ObjectReference, ctx context.Context, t *v1alpha1.Trigger) error {
+	gvk := schema.GroupVersionKind{
+		Group:   dependencyObjRef.GroupVersionKind().Group,
+		Version: dependencyObjRef.GroupVersionKind().Version,
+		Kind:    dependencyObjRef.Kind,
+	}
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	_, lister, err := r.buildInformerFactory.Get(gvr)
+	if err != nil {
+		logging.FromContext(ctx).Error(fmt.Sprintf("Error getting a lister for a resource with gvk '%+v', gvr '%+v'", gvk, gvr), zap.Error(err))
```

Comment:
Instead of using `fmt.Sprintf`, you can use `zap.Any`:

```go
logging.FromContext(ctx).Error("Error getting a lister for dependency resource", zap.Any("gvk", gvk), zap.Any("gvr", gvr), zap.Error(err))
```

Reply:
I agree with @grantr, normally Sprintf inside a zap log is a bad sign.

However, more generally, I think we don't need this log line (instead use the suggestion below).